### PR TITLE
 Added seen phrases

### PR DESCRIPTION
**Pull request by starjardin**
Hi team.

Seen phrases feature was implemented. Please have a look and leave comments to improve it.

Github [link](https://github.com/onja-org/malagasy-rn-Thallium/tree/seen-phrases).
Ticket [link](https://app.clickup.com/t/35bj8h).

Refs: 35bj8h

Thanks.
Honey

### CODE REVIEW

File: src/screens/Learning.js [hunk 1]
```diff
@@ -49,9 +57,10 @@ export default ({
   const selectAnswerCallback = useCallback(
     item => {
       if (item.id === currentPhrase.id) {
-        // TODO add to learned
-      } else {
-        // TODO add to seen
+        //TODO: add to learned
+        removeCorrectSeenPhrase(item);
+      } else if (!seenPhrases.some(el => el.id.includes(item.id))) {
```
Comment (vakodrazan) [inline src/screens/Learning.js]: The item.id repeats twice. It would be a good idea to create a variable for it.

File: src/screens/Home.js [hunk 2]
```diff
@@ -113,12 +123,17 @@ export default ({
...
+                id: '###seen-phrases###',
+                name: `${seenPhrases?.length} words and phrases`,
+              },
+            ]}
             text={'Learn'}
             color="#06B6D4"
             iconType="material-community"
             iconName="arrow-right"
-            makeAction={() => {}}
+            makeAction={seenPhrases?.length ? openSeenPhrases : () => {}}
```
Comment (vakodrazan) [inline src/screens/Home.js]: Create a variable for the repeated `seenPhrases?.length`

**Review by Ny-Rianala — APPROVED** [review]

File: src/screens/Home.js (hunk 2 above)
Comment (starjardin) [reply src/screens/Home.js]: Thanks, I have changed it.

**Review by vakodrazan — APPROVED** [review]

File: src/screens/Home.js [hunk 3]
```diff
@@ -63,6 +66,11 @@ export default ({
   const openNewTerms = () => {
     navigation.navigate('Add');
   };
+  const openSeenPhrases = () => {
```
Comment (andrzej-onja) [inline src/screens/Home.js]: if(!seenPhrases?.length) {
return
}

File: src/screens/Home.js (hunk 2 above)
Comment (andrzej-onja) [reply src/screens/Home.js]: move condition to the handler as in the comment, feels like its easier to read this way

File: src/screens/Learning.js [hunk 4]
```diff
@@ -133,7 +151,11 @@ export default ({
           </View>
           <View style={styles.heading}>
             <SectionHeading text="Category: " />
-            <Text>{currentCategoryName}</Text>
+            <Text>
+              {openSeenPhrases
```
Comment (andrzej-onja) [inline src/screens/Learning.js]: replace this with a check if current category is ##seen-phrases##.

You can set it in home `openSeenPhrases`

Comment (andrzej-onja) [reply src/screens/Learning.js]: remove openSeenPhrases and setOpenSeenPhrases from redux etc

**Review by andrzej-onja — CHANGES_REQUESTED** [review]

File: src/redux/selectors/index.js [hunk 5]
```diff
@@ -1,6 +1,8 @@
 import {createSelector} from 'reselect';
 
 export const categoriesRoot = state => state.categories;
+export const seenPhrasesRoot = state => state.seenPhrases;
+export const openSeenPhrasesRoot = state => state.openSeenPhrases;
```
Comment (andrzej-onja) [inline src/redux/selectors/index.js]: this was suppose to be removed

**Review by andrzej-onja — APPROVED** [review]

**Pull request merged**